File: src/main/java/social/alone/server/interest/InterestDto.java
package social.alone.server.interest;

import lombok.*;

import javax.validation.constraints.Size;

@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = {"value"})
public class InterestDto {

  public static InterestDto of(String value) {
    return new InterestDto(value);
  }

  @Size(min = 1, max = 30)
  String value;
}

File: src/main/kotlin/social/alone/server/interest/InterestDto.kt
package social.alone.server.interest

import lombok.EqualsAndHashCode
import javax.validation.constraints.Size

@EqualsAndHashCode(of = ["value"])
data class InterestDto(
        @Size(min = 1, max = 30)
        val value: String
) {

    companion object {

        fun of(value: String): InterestDto {
            return InterestDto(value)
        }
    }
}
